packages/@sfx/core/src/plugin.ts

```diff
@@ -51,6 +51,15 @@ export interface Plugin {
    * been initialized and may safely use other plugins.
    */
   ready?: () => void;
+
+  /**
+   * The callback for the unregistration phase of the lifecycle. The
+   * plugin is expected to perform teardown tasks in this function.
+   *
+   * In this function, the plugin can assume that other plugins are
+   * still available for use.
```

Comment:
While it's possible that plugin A has been torn down before plugin B, plugin A will still be available for plugin B. @DanielMoniz-GBI and I determined that this will be safe for most cases and that it is good enough for now.